pkg/auto/history/config/root.go

```diff
@@ -23,6 +24,13 @@ type Source struct {
 	Trait trait.Name `json:"trait,omitempty"`
 	// ReadMask instructs the history service to only read the specified fields
 	ReadMask *FieldMask `json:"readMask,omitempty"`
+	// Sample controls whether periodic downsampling is applied to the data source.
+	// It is intended for high-frequency sources where storing every data point is excessive.
+	// When Sample is set, the source is polled using a Cron schedule, and only values
+	// that differ from the previous sample are recorded — reducing storage by skipping redundant data.
+	// If Sample is not set (the default) or disabled, the source will be treated as event-driven:
+	// all changes will be recorded at their native frequency, with no sampling applied.
+	Sample *Sampling `json:"sample,omitempty"`
```

Comment:
Can we not use the term Sampling here. Sampling usually means to selectively choose records from a stream of them. That's not what you're doing here. It's worse in this case, too, because the normal behaviour _is_ a stream.